spec/acceptance/trusted_external_data_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'spec_helper_acceptance'
+
+context 'trusted external data' do
+  # Set up the trusted external command on the master
+  #
+  # TODO: Should this move to suite-level or even Rakefile-level
+  # setup? Because other tests like the classification test will
+  # also depend on it so running this all the time can get annoying.
+  before(:all) do
+    # TODO: Should this move to setup?
+    inventory_hash = inventory_hash_from_inventory_file
+    servicenow_instance_uri = on_servicenow_instance(&:uri)
+    servicenow_bolt_config = LitmusHelpers.config_from_node(inventory_hash, servicenow_instance_uri)
+    servicenow_config = servicenow_bolt_config['remote']
+    manifest = <<-MANIFEST
+class { 'servicenow_cmdb_integration::trusted_external_command':
+    instance => '#{servicenow_instance_uri}',
+    user     => '#{servicenow_config['user']}',
+    password => '#{servicenow_config['password']}',
+}
+    MANIFEST
+
+    on_master do |master|
+      master.idempotent_apply(manifest)
+    end
+  end
+
+  context '$trusted.external.servicenow hash' do
+    separator = '<trusted_json>'
+
+    before(:all) do
+      manifest = <<-MANIFEST
+$trusted_json = inline_template("<%= @trusted.to_json %>")
+notify { "trusted external data":
+  message => "#{separator}${trusted_json}#{separator}"
+}
+      MANIFEST
+      # TODO: This will cause some problems if we run the tests
+      # in parallel. For example, what happens if two targets
+      # try to modify site.pp at the same time?
+      set_sitepp_content(manifest)
+    end
+
+    after(:all) do
+      set_sitepp_content('')
+    end
+
+    it "contains the node's CMDB record" do
+      result = trigger_puppet_run(LitmusHelpers.instance)
+      puppet_output = result.stdout
+
+      trusted_json = nil
+      begin
```

Comment:
This block, and the process of getting the `cmdb_record_id` are a lot of code to put in an `it` block. Could these be extracted to a helper file so the test reads more naturally?

Reply:
Yeah, it does look like a lot of code and the Rake tasks are already getting the `cmdb_record_id`. I can do that.